src/Compilers/CSharp/Portable/FlowAnalysis/NullableWalker.cs

```diff
@@ -1233,7 +1230,7 @@ private void EnforceDoesNotReturn(SyntaxNode? syntaxOpt)
 
 #if DEBUG
             // https://github.com/dotnet/roslyn/issues/34993 Enable for all calls
-            if (compilation.NullableSemanticAnalysisEnabled)
+            if (compilation.IsNullableAnalysisEnabled)
```

Comment:
I tried testing for both properties (here and below) in an earlier commit, but it looks like callers are depending on `IsNullableAnalysisEnabled` since many tests failed. If we need this DEBUG verification for `IsNullableAnalysisEnabledAlways`, let's add that later.

---
In reply to: [542546400](https://github.com/dotnet/roslyn/pull/49876#discussion_r542546400) [](ancestors = 542546400)